docs/cloud-integration.md

```diff
@@ -190,6 +217,46 @@ while they are still being written. Applications can write straight to the monit
 atomic `rename()` operation.
 Otherwise the checkpointing may be slow and potentially unreliable.
 
+## Committing work into cloud storage safely and fast.
+
+As covered earlier, commit-by-rename is dangerous on any object store which
+exhibits eventual consistency (example: S3), and often slower than classic
+filesystem renames.
+
+Some object store connectors provide custom committers to commit tasks and
+jobs without using rename. In versions of Spark built with Hadoop-3.2 or later,
+the S3A connector for AWS S3 is such a committer.
+
+Instead of writing data to a temporary directory on the store for renaming,
+these committers write the files to the final destination, but do not issue
+the final POST command to make a large "multi-part" upload visible. Those
+operations are postponed until the job commit itself. As a result, task and
+job commit are much faster, and task failures do not affect the result. 
+
+To switch to the S3A committers, use a version of Spark which includes the
+Hadoop-3.1+ binaries, and switch the committers through the following
+options.
+
+```
+spark.hadoop.fs.s3a.committer.name directory
+spark.sql.sources.commitProtocolClass org.apache.spark.internal.io.cloud.PathOutputCommitProtocol
+spark.sql.parquet.output.committer.class org.apache.spark.internal.io.cloud.BindingParquetOutputCommitter
+```
+
+Normal dataframe write commands will then use this committer for any format
```

Comment:
This paragraph sounds a little self-contradictory.

- Committer will be used for any format that doesn't have a custom one
- These are the supported formats.

Does that mean that formats that aren't explicitly called out are not necessarily supported or tested or what?

(Removing the list of "supported formats" would help in avoiding the ambiguity.)

Reply:
changed to say : "output formats which are known to work (i.e. the ones tested)"

Reply:
I think that phrasing just raises question of "which are these formats". If you really insist on saying something, just say "tested with the most common formats supported by Spark".

Reply:
Done